libqtile/command/base.py

```diff
@@ -27,12 +27,42 @@
 import abc
 import inspect
 import traceback
-from typing import Callable, List, Optional, Tuple, Union
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Tuple, TypeVar, Union
 
 from libqtile.command.graph import SelectorType
 from libqtile.log_utils import logger
 
 ItemT = Optional[Tuple[bool, List[Union[str, int]]]]
+AnyClass = TypeVar("AnyClass")
+
+
+def _register_command(register: List) -> Callable:
+    """
+    Decorator function to register commands to be exposed to
+    command clients.
+    """
+    def wrapper(func) -> Callable:
+        # As decorators are processed before classes are instantiated,
+        # we cannot do a search of class methods when looking for available
+        # commands. Instead, we need to identify which module the unbound
+        # method belongs to.
+        # However, decorators are processed early and will happen before some
+        # modules are fully imported. So, we first register a tuple of
+        # (the module containing the class, the class name, the method).
+        # Once qtile is starting we call a separate method to generate
+        # references to the class type containing the function.
+        register.append(
+            (
+                inspect.getmodule(func),
+                # This takes the dotted name for the method and trims it
+                # to give the module.
+                func.__qualname__.split('.<locals>', 1)[0].rsplit('.', 1)[0],
```

Comment:
Got this line from StackOverflow. I'm pretty sure we don't need the `.split('.<locals>', 1)[0]` part as none of these methods should be in `locals`.